docs/ref/contrib/admin/index.txt

```diff
@@ -3208,6 +3208,33 @@ The presence of the ``admin_password_reset`` named URL will cause a "forgotten
 your password?" link to appear on the default admin log-in page under the
 password box.
 
+Creating custom checks and tests
+-------------------------------
+
+The global variable ``all_sites`` is a weak set of all ``AdminSite`` instances.
+You can iterate through ``all_sites`` to perform custom checks and tests.
+
+In this example, the ``test_all_sites_config`` function iterates over the
+``all_sites`` set and performs tests on each ``AdminSite`` instance to ensure 
+that its configuration properties are not ``None``. If any of the tests fail, 
+an ``AssertionError`` is raised, indicating that the ``AdminSite`` instance
+is not properly configured:
+
+.. code-block:: python
+    from django.contrib.admin.sites import AdminSite, all_sites
+
+
+    def test_all_sites_config():
+        """
+        A custom function that tests the configuration of all `AdminSite` instances
+        """
+
+        for site in all_sites:
+            # Test that the `AdminSite` is properly configured
+            assert site.site_title is not None
+            assert site.site_header is not None
+            assert site.index_title is not None
+
```

Comment:
IMO the extensive paragraph is not needed. I'd add a small reference to the `AdminSite` docs, e.g.
```diff
diff --git a/docs/ref/contrib/admin/index.txt b/docs/ref/contrib/admin/index.txt
index 3fd13b28e8..2eb5565fcb 100644
--- a/docs/ref/contrib/admin/index.txt
+++ b/docs/ref/contrib/admin/index.txt
@@ -2831,6 +2831,10 @@ linked to the document in ``{% block dark-mode-vars %}``.
     See :ref:`customizing-adminsite` for an example of customizing the
     :class:`AdminSite` class.
 
+.. data:: django.contrib.admin.sites.all_sites
+
+    A weak set contains all admin site instances.
+
 ``AdminSite`` attributes
 ------------------------
 

```

@adamchainz What do you think?

Reply:
Yes something like that, but with a link to `WeakSet` in the python docs, since it's a rare data type.

```diff
diff --git a/docs/ref/contrib/admin/index.txt b/docs/ref/contrib/admin/index.txt
index 3fd13b28e8..2eb5565fcb 100644
--- a/docs/ref/contrib/admin/index.txt
+++ b/docs/ref/contrib/admin/index.txt
@@ -2831,6 +2831,10 @@ linked to the document in ``{% block dark-mode-vars %}``.
     See :ref:`customizing-adminsite` for an example of customizing the
     :class:`AdminSite` class.
 
+.. data:: django.contrib.admin.sites.all_sites
+
+    A :class:`~weakref.WeakSet` contains all admin site instances.
+
 ``AdminSite`` attributes
 ------------------------
```

(untested, `:class:` should work though...)